extensions/agroal/deployment/src/main/java/io/quarkus/agroal/deployment/AgroalProcessor.java

```diff
@@ -100,8 +103,12 @@ BeanContainerListenerBuildItem build(
                 java.sql.ResultSet.class.getName(),
                 java.sql.ResultSet[].class.getName()));
 
-        validateBuildTimeConfig(null, agroalBuildTimeConfig.defaultDataSource);
-        agroalBuildTimeConfig.namedDataSources.forEach(AgroalProcessor::validateBuildTimeConfig);
+        validateBuildTimeConfig(null, agroalBuildTimeConfig.defaultDataSource, deploymentClassLoader.getClassLoader());
+        Map<String, DataSourceBuildTimeConfig> namedDataSources = agroalBuildTimeConfig.namedDataSources;
```

Comment:
It's done, it's done but lambdas are usually accepted at build time (unless it doesn't make the code clearer) but in this case, it was a bit more readable before.

Anyway, it's no biggie, let's keep it the way you did it.

Reply:
I have developed lambda aversion 😂